keps/0012-label-index.md

```diff
@@ -0,0 +1,51 @@
+---
+kep-number: 12
+title: Index labels to support efficient queries by label selectors
+authors:
+  - "@jian-he, @ccding"
+owning-sig: sig-api-machinery
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: "2018-05-31"
+last-updated: "2018-05-31"
+status: provisional
+---
+
+## Summary
+
+This proposal presents solutions to make the queries by label selectors in the
+list API efficient.
+
```

Comment:
To make it clear, this proposal focus on client cache or apiserver watch cache?

Reply:
See Proposal section 1 and 4. It focuses on client cash, while apiserver watch cache is optional.

Reply:
These two things shouldn't be in the same KEP, they don't have much in common.

Reply:
Removed the apiserver side cache. We will do it in the client side first.